src/types/main.d.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * For default interfaces that are frequently used and extensions:
+ *   - HTMLElement: h5s and its inner objects.
+ *   - Extension of objects that are external to the library.
+*/
+
+interface opts {
+    connectWith: boolean,
+    acceptFrom: void,
+    copy: boolean,
+    placeholder: void,
+    disableIEFix: boolean,
+    placeholderClass: string,
+    draggingClass: string,
+    hoverClass: boolean,
+    debounce: number,
+    maxItems: number,
+    itemSerializer: void,
+    containerSerializer: void,
+    items: string
+}
+
+// Can these functions be better defined?
+interface events {
+    [index:string]: Function
+    dragStart: () => void,
+    dragEnd: () => void,
+    drop: () => void,
+    dragover: () => void,
+    dragenter: () => void
+}
+
+
+interface data {
+    [index:string]: string|opts,
+    opts: opts,
+    _disabled: string,
+    items: string,
+    connectWith: string
+}
+
+interface h5s {
+    events: events,
+    data: data
+}
+/* 
+ * typescriptlang.org/docs/handbook/declaration-merging.html
+ * ie. if this is done new properties can be 
+ * added to previously defined objects
+*/
+interface HTMLElement {
```

Comment:
I think we should not extend `HTMLElement`. Rather, in the future we should have an interface for the sortable class #326

Reply:
I can look into that next